### PR TITLE
[tests-only][full-ci] Refactor acceptance sharing tests

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Web issue https://github.com/owncloud/web/issues/5857 has been fixed and the tests related to this issue should pass.
While looking through those tests, I found that there are some unnecessary test steps and broken step implementations.
This PR has a refactoring of those tests.

## How Has This Been Tested?
- test environment: local

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroups</strong> https://drone.owncloud.com/owncloud/web/24405/27/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroups-shareWithGroups_feature-L166.png
  


</p></details>

File: tests/acceptance/features/webUISharingFilePermissionMultipleUsers/shareFileWithMultipleUsers.feature [hunk 1]
```diff
@@ -10,29 +10,18 @@ Feature: Sharing files with multiple internal users with different permissions
...
     And the user selects the following collaborators for the share as "<role>" with "<extra-permissions>" permissions:
       | collaborator | type |
-      | Regular User | user |
       | Brian Murphy | user |
       | Carol King   | user |
-      | David Lopez  | user |
-    And the user removes "David Lopez" as a collaborator from the share
-    And the user removes "Regular User" as a collaborator from the share
-    Then user "David Lopez" should not be visible in the collaborators selected options in the webUI
-    And user "Regular User" should not be visible in the collaborators selected options in the webUI
```
Comment (saw-jan) [inline tests/acceptance/features/webUISharingFilePermissionMultipleUsers/shareFileWithMultipleUsers.feature:34]: The scenario is to share a file with multiple users so IMO these steps are irrelevant. I have added a new scenario to check the remove functionality of collaborators from the invite list

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 2]
```diff
@@ -678,7 +698,7 @@ module.exports = {
     },
     groupInSelectedCollaboratorsList: {
       selector:
-        '//span[contains(@class, "files-share-invite-recipient")]//span[.="Group"]/following-sibling::p[.="%s"]',
+        '//span[contains(@class, "files-share-invite-recipient")]//span[@data-testid="recipient-icon"]/following-sibling::p[.="%s"]',
```
Comment (saw-jan) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js:701]: This selector was broken

**Comment by kiranparajuli589** [comment]
lgtm :+1:

**Review by fschade — APPROVED** [review]
LGTM, thanks

**Pull request merged**